pyodide_build/mkpkg.py

```diff
@@ -71,20 +75,54 @@ def make_package(package):
         yaml.dump(yaml_content, fd, default_flow_style=False)
 
 
+def update_package(package):
+    import yaml
+
+    with open(PACKAGES_ROOT / package / 'meta.yaml', 'r') as fd:
+        yaml_content = yaml.load(fd, Loader=yaml.FullLoader)
+
+    if set(yaml_content.keys()) != set(('package', 'source', 'test')):
+        print(
+            'Only pure-python packages can be updated using this script. '
+            'Aborting.')
+        sys.exit(1)
+
+    sdist_metadata, pypi_metadata = _get_metadata(package)
+    pypi_ver = pypi_metadata['info']['version']
+    local_ver = yaml_content['package']['version']
+    if pypi_ver <= local_ver:
+        print(f'Already up to date. Local: {local_ver} Pypi: {pypi_ver}')
+        sys.exit(0)
+    print(f'Updating {package} from {local_ver} to {pypi_ver}')
+
+    if 'patches' in yaml_content['source']:
+        import warnings
+        warnings.warn(
+            f"Pyodide applies patches to {package}. Update the "
+            "patches (if needed) to avoid build failing.")
+
+    yaml_content['source']['url'] = sdist_metadata['url']
+    yaml_content['source'].pop('md5', None)
+    yaml_content['source']['sha256'] = sdist_metadata['digests']['sha256']
+    yaml_content['package']['version'] = pypi_metadata['info']['version']
+    with open(PACKAGES_ROOT / package / 'meta.yaml', 'w') as fd:
+        yaml.dump(yaml_content, fd, default_flow_style=False)
+
+
 def make_parser(parser):
-    parser.description = '''
-Make a new pyodide package. Creates a simple template that will work
-for most pure Python packages, but will have to be edited for more wv
-complex things.'''.strip()
+    parser.description = '''Add or update a python package in pyodide.'''
+    parser.add_argument(
+        'package', type=str, nargs=1, help="The package name on PyPI")
     parser.add_argument(
-        'package', type=str, nargs=1,
-        help="The package name on PyPI")
+        '--update', action='store_true', help='update existing package')
```

Comment:
Instead of having a flag, would it make sense to:

1) Check if the file already exists, if so update
2) If not, create from scratch

?

Reply:
That's a valid solution as well. My solution is more explicit, which probably doesn't mater here. The code would be simpler with your solution.

I could make the change tomorrow afternoon if you'd prefer it.

Reply:
I'd like to leave this as-is, with the happy side-effect of a "good first issue" created to improve combine these 2 flows.